Doc/reference/compound_stmts.rst

```diff
@@ -192,9 +192,8 @@ those made in the suite of the for-loop::
 
 Names in the target list are not deleted when the loop is finished, but if the
 sequence is empty, they will not have been assigned to at all by the loop.  Hint:
-the built-in function :func:`range` returns an iterator of integers suitable to
-emulate the effect of Pascal's ``for i := a to b do``; e.g., ``list(range(3))``
-returns the list ``[0, 1, 2]``.
+the built-in type :func:`range` represents an immutable sequence of integers from 0
+to a specified number; e.g., ``list(range(3))`` returns the list ``[0, 1, 2]``.
```

Comment:
Yes, dump the Pascal reference.  I suggest an even better replacement.
```suggestion
the built-in type :func:`range` represents immutable arithmetic sequences of integers.  For instance, iterating ``range(3)`` successively yields 0, 1, and then 2.
```
Types represent the concept that covers all possible instances, not any one instance.  In the context of looping, the iteration behavior of ranges is relevant, not what the list constructor does with one.